### ISSUE TITLE
Modular Birdshot

### ISSUE THREAD

**Issue by Zytolg** [tgstation/tgstation#83603]
As the name implies. The idea was actually brought up by @Maurukas and I liked it. The station is still receiving it's V2 passes for the baseline station, but at some point, I'd like to make certain sections of the station modularization as described below. 

- [ ] - Modular Enigneering
Basically medium% spinoffs of the engineering satellite. I'd like these to focus on specializing a main function. In one version engineering has decided to expand the Atmospherics Workshop, and in the other, the Super Matter Engine is a behemoth of a machine. I think mixing in a degree of uncertainty can really add some engagement to otherwise static game-play features. This won't be everyone's cup of tea, but I also think this is the next step in developing the station's engineering department. 

- [ ] Modular Prison
Same principals as the Modular Engineering Task mentioned above. Instead we focus around the prison satellite. Currently, the Birdshot prison doesn't have anything going on. I believe the Security Department (Brig) to be 100% functional and in the least need of a rework, but I also think the Prison has room to grow. Instead of doing the same thing again, but with better looks, introducing molecularity to the Birdshot Prison would be means to encourage further interaction with the area by Security and Prisoner Players (Sec should be setting up solars in this area but they never do). I have three concepts I want to explore: Escape, Fortress, and Subject. 

- [ ] Modular Dorms
This is the only item on this list that isn't able to be worked on at the time of writing. V2 Birdshot will feature dorms as its own satellite not too dissimilar from what Engineering is. I can't go into details because I haven't figured out what's going to be on it outside of Dorms and the Arrivals Berth, but I know at this point in time that I would like it to be modular. I'll update this section of the milestone when Birdshot V2 is closer to completion. 

- [ ] Post Modular Birdshot Updates
In short, other sections of Birdshot might receive a Modularity pass in the future, but these will be the most difficult to maintain given their relationship to other spaces on the station. I want to keep the possibility open to explore, but only after the 3 points above are added to the station. 

This idea is pretty much fresh off the "hey this is actually a novel idea" forge, so I haven't put that much else into it. It'll be something else to maintain, but again, it opens up a lot of future potential for the station and its game-play. If anyone wants help by starting work on Modular Engineering or Modular Prison, reach out and I can help you get that ball rolling. I'll update this Milestone after I've had more time to work out the finer details in my head.

**Comment by MrMelbert** [comment]
FTR I don't really want any map to have large sections of a department be using the modular mapping system, I think it's too unreliable (particularly for reviewing (checking all possible combinations is too time consuming), testing (we can't unit test all possible combinations feasibly), and maintainability (needing to change a segment is far too time consuming, as someone who had to go through the tram modules))

**Comment by Maurukas** [comment]
Right now we have a major feature in modular mapping that's very under utilized. A tiny fraction of it's potential is used in exactly one map, one ruin, and in a way that doesn't bring that feature to prominence. There's been very little done to improve the feature since it was added to the game, and why bother since it's only used on two maps? It's not supported in sDMM, it requires clunky toml files, the tooling doesn't like it. Why spend time improving it? It's not used anywhere. Why use it somewhere? It's clunky and the tooling sucks. It's a vicious cycle.

We also don't use it to its full potential. It is used exclusively for swapping in and out large chunks over blank sections of map, but it doesn't have to be. You can use it as an overlay with pass through turfs, painting a set of changes over top an existing map structure - this allows a lot more variability than you can get with randomized spawners.

I'm of the opinion that we should use the feature more, and I'm willing to trade off at least a little maintainability in the name of doing something cool and unique in game. Where exactly that line should be drawn is open to discussion and this is a good place for it.

I think the prison and the dorms areas are great places to start. These are not weight bearing parts of the station. Issues here will have limited impact on the rest of the crew, and these areas are essentially aesthetic. This is where modularity has the best chance of succeeding.

Engineering is probably not the best place to go with this, at least not with places that house important equipment. This can have a pretty severe impacts on the round, especially after recent power changes.

If it proves unmaintainable, or problematic, then we can change things. Nothing is permanent.

**Comment by MrMelbert** [comment]
I'm not saying it's unusable because it's clunky, but because it's just nightmarish to envision having to maintain every single module of a department, especially when that's the big hold-up with maintaining more and more maps

You are a god of map maintaining Vire but the average person is not on your level and pushing this to them will cause suffering

**Comment by LemonInTheDark** [comment]
This scares me. Modular maps have fucked me up in the past and this'd be wayyy worse.

Also, the issue tracker is not a project planning section. If you want that use the player dev projects channel on discord. Closing for this reason
